src/core/action-creator.ts

```diff
@@ -47,10 +47,16 @@ export function createActions(flux: FluxCapacitor) {
           {
             ...metadata,
             validator: {
-              payload: {
-                func: ({ query }) => !('query' in search) || !!query || query === null,
-                msg: 'search term is empty'
-              }
+              payload: [
+                {
+                  func: ({ query }) => !('query' in search) || !!query || query === null,
+                  msg: 'search term is empty'
+                },
```

Comment:
typically we try to keep the brackets on the same line for closing one object and opening the next
ex: `}, {`